### PR TITLE
DOC: Clarify Generator docs in ``np.random``.

### LINKED ISSUE #17140: DOC: redirect notice to `default_rng()` is not linked
From [this documentation page](https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html)

image

None of that message is actually clickable.

### PR DESCRIPTION
**Pull request by kubekmonika**
I changed the descriptions so that now:
* they say about `Generator` instance and not `default_rng()` instance, as the later is only a class constructor,
* under See Also section one can find a link to the recommended `Generator` constructor.

Closes #17140

### CODE REVIEW

File: numpy/random/mtrand.pyx [hunk 1]
```diff
@@ -403,6 +403,7 @@ cdef class RandomState:
         See Also
         --------
         Generator.random: which should be used for new code.
+        Generator.default_rng: Recommended constructor for `Generator`.
```
Comment (rkern) [inline numpy/random/mtrand.pyx:406]: This is not needed for these methods.

**Review by rkern — CHANGES_REQUESTED** [review]
You can remove all of the lines referring to `default_rng` in the `See Also` sections. That's not particularly germane to each of these methods

**Comment by rkern** [comment]
The original bug report #17140 was more about the fact that the referred-to method was not a clickable link (and secondarily that it ought not to refer to the non-existent "`default_rng` instances"). You'll have to use single-backticks to get the links. I would probably write those sentences like so:

```
            New code should use the `Generator.random` method instead; please
            see the :ref:`random-quick-start`.
```

I see in that issue that it was recommended that `default_rng` be added to the `See Also` list, but I think that is not a good idea. The reference to the `Quick Start` material in the note is more useful.

In order to make the methods in the `See Also` list clickable, there needs to be a space before the `:`. The ReST parser treats those two cases differently. That would be helpful to close #17140 fully.

**Comment by rkern** [comment]
Actually, the full syntax for the `note` should look like this, in order to fix the links:

```
            New code should use the :meth:`~numpy.random.Generator.random`
            method of a :class:`~numpy.random.Generator` instance instead;
            please see the :ref:`random-quick-start`.
```

The default scoping for these method docstrings look up only _under_ `RandomState` rather than up to `numpy.random`, unfortunately. So we need to use the fully-qualified `numpy.random.Generator` in order to locate the correct referent. Using the `~` makes the generated HTML only use the part after the last `.` as the link text, though, so this will read as "use the `random` method of a `Generator` instance" with the correct links.

Similarly, the `See Also` section needs fully-qualified references in addition to the spaces before and after the colon:

```
        See Also
        --------
        numpy.random.Generator.random : which should be used for new code.
```

**Comment by melissawm** [comment]
Hello, @kubekmonika - are you interested in picking up this PR again? If you need help with it feel free to ping me. Thanks!

**Comment by charris** [comment]
This looks pretty straight forward, a few fixes should make it ready to go. The conflicts are due to `default_rng` being removed from the `See Also` section and should be easy to fix.

**Comment by bsipocz** [comment]
I've rebased this PR to resolve the conflicts and finished addressing the review items in #22304, closing this one now (the referenced, original issue has been addressed in the meantime and is now closed, too)


Thank you @kubekmonika for your contribution, your code changes have been also moved to the other PR.

**Pull request closed without merging**